src/query/ts/metadata.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ts
+
+// MetricType is the enum for metric types.
+type MetricType int
+
+const (
+	// MetricTypeCounter is the counter metric type.
+	MetricTypeCounter MetricType = iota
+
+	// MetricTypeGauge is the gauge metric type.
+	MetricTypeGauge
+
+	// MetricTypeTimer is the timer metric type.
+	MetricTypeTimer
+)
+
+// SourceType is the enum for metric source types.
+type SourceType int
+
+const (
+	// SourceTypePrometheus is the prometheus source type.
+	SourceTypePrometheus SourceType = iota
+
+	// SourceTypeGraphite is the graphite source type.
+	SourceTypeGraphite
+)
+
+// SeriesInfo is information about the time series.
+type SeriesInfo struct {
```

Comment:
nit: SeriesAttributes or SeriesMetdata could also be alternative names, we already have `storage.Attributes` for each write (retention/resolution). What do you think? Not super bullish, but info always seems so generic.

Reply:
`SeriesAttributes` is good. Will change to that.